compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -152,15 +166,32 @@ private String getName(TypeMirror type) {
             .build())
         .addModifiers(PUBLIC, FINAL)
         .addSuperinterface(ClassName.get("bullet", "ObjectGraph"))
-
         .addField(elementName, "component", PRIVATE, FINAL)
-
         .addMethod(MethodSpec.constructorBuilder()
             .addModifiers(PUBLIC)
             .addParameter(elementName, "component", FINAL)
             .addCode("this.component = component;\n")
             .build());
 
+    if (membersInjectionMethodsMap.size() > 0) {
+      classBuilder.addField(ClassName.get("bullet.impl", "ClassIndexHashTable"), "classIndexHashTable", PRIVATE, STATIC, FINAL);
+    }
+
+    // Generate the ClassIndexHashTable if there are classes to inject.
+    if (membersInjectionMethodsMap.size() > 0) {
+      // ClassIndexHashTable size should be at least 30% larger and a prime number.
+      int classIndexHashTableSize = getNextPrime((int) Math.ceil(membersInjectionMethodsMap.size() * 1.3));
```

Comment:
To be precise (and I realize it was me that wasn't precise) you want to achieve a maximum of 70% load factor. Multiplying by 1.3 can be 77% load factor. What I do in my own implementation is divide by 0.7 to get at least a 70% load factor (or you could multiply by 1.43).